src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java

```diff
@@ -279,6 +271,12 @@ public void run() {
             } finally {
                 cacheLock.unlock();
             }
+            // close connections outside cacheLock
+            if (closeList != null) {
```

Comment:
Should we perhaps move this entire `if` block inside the `finally` just after the call to `cacheLock.unlock()`. That way even if there was any exception in that loop, then we wouldn't end up losing and thus leaking these unclosed sockets?